### DISCUSSION TITLE
Unable to copy / paste into remote terminal

### DISCUSSION THREAD

**Discussion by ghost** [isontheline/pro.webssh.net#875] [General]
I am unable to copy / paste to a remote terminal. Clipboard strategy is set for allow read / write.

Also, another oddity I've discovered is that with some remote terminals that up-arrow history doesn't work. 

Any suggestions I can try?

**Comment by isontheline** [comment]
Which platform are you using?

**Reply by ghost** [reply]
Mac store version, latest.

**Reply by isontheline** [reply]
That's strange copy / paste should work out of the box, I use it hourly.

> Clipboard strategy is set for allow read / write

This strategy is only about accessing the clipboard by the remote application itself, not the "general pasteboard" you use for input / output.

Latest trouble with the macOS clipboard is here : https://github.com/isontheline/pro.webssh.net/discussions/676

But I didn't hear to new problems since.

Have you tried to restart WebSSH / device?

**Reply by ghost** [reply]
It's all a bit odd...

I use [coderunner](https://coderunnerapp.com) to store scripts etc and I can copy / paste  in both directions to / from other apps like Apple notes, Pages etc but not to WebSSH.

With WebSSH I can copy / paste to and from other apps AND also coderunner.

**Reply by isontheline** [reply]
Thanks for the hint about coderunner!

I will try to find the root cause. Perhaps the clipboard is filled with enhanced with metadata that WebSSH doesn't read by default.

I will make a try tonight (Europe/Paris)

**Reply by ghost** [reply]
Many thanks (Europe / Les Carroz :P )

**Reply by ghost** [reply]
Another one just happened now - connected to a digital ocean droplet and copy a line from the screen output and pasting back into the terminal prompt (**apt install docker.io**) copies but doesn't paste.

**Reply by isontheline** [reply]
Copy and paste have been triggered from the same terminal session on WebSSH?

I ask because I have an another open issue open about a bad focus management on my own. Sometimes the terminal needs to be focused again (left click) before to be able to interact with the keyboard.

**Reply by ghost** [reply]
Yes, same terminal session.

**Reply by isontheline** [reply]
Sometimes when I copy from "Microsoft Word" app I encounter your paste problem.
I'm opening an issue to work on this trouble : #877

I hope to release a fix on April, Sunday 30th
